instrumentation/apache-httpclient/apache-httpclient-5.2/library/src/main/java/io/opentelemetry/instrumentation/apachehttpclient/v5_2/internal/Experimental.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.apachehttpclient.v5_2.internal;
+
+import io.opentelemetry.instrumentation.apachehttpclient.v5_2.ApacheHttpClientTelemetryBuilder;
+import java.util.function.BiConsumer;
+import javax.annotation.Nullable;
+
+/**
+ * This class is internal and experimental. Its APIs are unstable and can change at any time. Its
+ * APIs (or a version of them) may be promoted to the public stable API in the future, but no
+ * guarantees are made.
+ */
+public final class Experimental {
+
+  @Nullable
+  private static volatile BiConsumer<ApacheHttpClientTelemetryBuilder, Boolean>
```

Comment:
> shouldn't also be needed for correctness as this field is set from the static initializer of `ApacheHttpClientTelemetryBuilder`

oh, even if the static field is in a different class than the static initializer?

Reply:
I don't think that the barriers care where the fields are, so that at the end of `ApacheHttpClientTelemetryBuilder` class initializer everything that happened in any class should be visible. Also the `setEmitExperimentalTelemetry` needs an `ApacheHttpClientTelemetryBuilder` which requires that the class is run.